### PR TITLE
hlcan [canif-name] mod

### PR DESCRIPTION
**Pull request by Izan00**
This code allows changing the default device name from can0 to any name while defining the can connection.

### CODE REVIEW

**Comment by alexmohr** [comment]
Looks great, I'll test it in the next couple of days and merge it then

**Comment by alexmohr** [comment]
Please also adjust the help printout to document how this function works. I also would prefer if you could add an a command line switch for this

File: src/hlcand.c [hunk 1]
```diff
@@ -326,7 +332,36 @@ int main(int argc, char *argv[])
 		exit(EXIT_FAILURE);
 	}
 	
+	/* retrieve the name of the created CAN netdevice */
+	if (ioctl(fd, SIOCGIFNAME, ifr.ifr_name) < 0) {
```
Comment (alexmohr) [inline src/hlcand.c:336]: This is not really working and the application always exits in line `338` no matter if a name was given or not. 
You also don't need it as `ifr.ifr_name` already contains `can0` (for example), it's also logged in `342`.
If the `if` is removed the application will exit in `342` with `invalid ioctl for device`. 

Please do not blindly copy from https://github.com/linux-can/can-utils/blob/master/slcan_attach.c#L225 and test your changes. 
Furthermore this change introduces a kernel freeze in some scenarios I haven't debugged. 

Let me know if you need further help to implement this feature.

**Comment by alexmohr** [comment]
If you need this feature please let me know. I can help you implementing it or implement it myself.

**Comment by Izan00** [comment]
Hi, the adapter was broken during the shipping, thus it didn't work with your code neither, but the rtx led was flashing, so I assumed it was ok. At this moment I am not planning to spend more time with hlcan as I have another can adapter working. Thanks for your time, maybe in the future I'll try it again with a new one.

**Pull request closed without merging**